src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/StringBuilderAppendShouldNotTakeSubstring.Fixer.cs

```diff
@@ -88,6 +92,23 @@ public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
             return document.WithSyntaxRoot(newRoot);
         }
 
+        private struct FixComponents
```

Comment:
Looks like this can be a `readonly struct`.

Reply:
could be, but the projects are on C#7, so no readonly struct available at all.